packages/catalog-client/api-report.md

```diff
@@ -189,6 +199,42 @@ export interface GetEntityFacetsResponse {
   >;
 }
 
+// @alpha
+export type GetPaginatedEntitiesCursorRequest = {
+  fields?: string[];
+  limit?: number;
+  cursor: string;
+};
+
+// @alpha
+export type GetPaginatedEntitiesInitialRequest =
+  | {
+      fields?: string[];
+      limit?: number;
+      filter?: EntitiesFilter;
+      sortField?: string;
+      query?: string;
+      sortFieldOrder?: 'asc' | 'desc';
+    }
+  | undefined;
+
+// @alpha
+export type GetPaginatedEntitiesRequest =
+  | GetPaginatedEntitiesInitialRequest
+  | GetPaginatedEntitiesCursorRequest;
+
+// @alpha
+export type GetPaginatedEntitiesResponse = {
+  entities: Entity[];
+  totalItems: number;
+  next?(
+    request?: Omit<GetPaginatedEntitiesCursorRequest, 'cursor'>,
+  ): Promise<GetPaginatedEntitiesResponse>;
+  prev?(
+    request?: Omit<GetPaginatedEntitiesCursorRequest, 'cursor'>,
+  ): Promise<GetPaginatedEntitiesResponse>;
+};
```

Comment:
Yeah I get your point about serialization. I wanted to give the users an easier way to query the next/prev batch. Regarding the serialization issue you’ve mentioned I’m seeing that I made a mistake here, without including the cursor in the response, which is going to be needed for example in case we want to save the cursor as queryparam, for example. So, I agree, I will remove the two next/prev utility methods.

> I'm thinking with that we might be able to call this one listEntity or queryEntities or something like that

yep, I'm very open to using one of the two suggestions. I think adding this logic to `getEntities` might be hard, since it takes different arguments and a different response. But maybe it is possible using some method overloading. I will check.